Update the entire Readme (#90)
spree/spree-storefront-api-v2-js-sdk: README.md

})
```

### `removeItem`
Removes Line Item from Cart.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Parameters schema:__
```ts
line_item_id: string
```

__Success response schema:__ [Success schema](#success-schema)

__Failed response schema:__ [Error schema](#error-schema)

__Example:__

```ts
// Logged in user
const response = await client.cart.removeItem({ bearerToken }, '1')

// or guest user
const response = await client.cart.removeItem({ orderToken }, '1')
```

### `emptyCart`
Empties the Cart.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Success response schema:__ [Success schema](#success-schema)

__Failed response schema:__ [Error schema](#error-schema)

__Example:__

```ts
// Logged in user
const response = await client.cart.emptyCart({ bearerToken })

// or guest user
const response = await client.cart.emptyCart({ orderToken })
```

### `applyCouponCode`
Applies a coupon code to the Cart.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Parameters schema:__
```ts
{
  coupon_code: string
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Loged in user
const response = await client.cart.applyCouponCode({ bearerToken }, {
  coupon_code: 'promo_test'
})

// or guest user

> const response = await client.cart.applyCouponCode({ orderToken  }, {

Reviewer: ```suggestion
const response = await client.cart.applyCouponCode({ orderToken }, {
```

  coupon_code: 'promo_test'
})
```

### `removeCouponCode`
Removes a coupon code from the Cart.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Parameters schema:__
```ts
coupon_code: string
```

__Success response schema:__ [Success schema](#success-schema)

> __Filed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.removeCouponCode({ bearerToken }, 'promo_test')

// or guest user
const response = await client.cart.removeCouponCode({ orderToken }, 'promo_test')
```

### `estimateShippingMethods`
Returns a list of Estimated Shipping Rates for Cart.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Parameters schema:__
```ts
country_iso: string
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.estimateShippingMethods({ bearerToken }, 'USA')

// or guest user
const response = await client.cart.estimateShippingMethods({ orderToken }, 'USA')
```

## [Checkout](https://guides.spreecommerce.org/api/v2/storefront/#tag/Checkout)

### `orderUpdate`

> Updates the Checkout

Reviewer: ```suggestion
Updates an Order. It's possible to perform multiple updates on the same Order and provide different data types and values each time.
```

You can run multiple Checkout updates with different data types.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Parameters schema:__
```ts
order: {

>   email: string

Reviewer: ```suggestion
  email?: string
```

  bill_address_attributes?: {
    firstname: string
    lastname: string
    address1: string
    city: string
    phone: string
    zipcode: string
    state_name: string
    country_iso: string
  }
  ship_address_attributes?: {
    firstname: string
    lastname: string
    address1: string
    city: string
    phone: string
    zipcode: string
    state_name: string
    country_iso: string
  }
  shipments_attributes?: [
    {
      selected_shipping_rate_id: number
      id: number
    }
  ]
  payments_attributes?: [
    {
      payment_method_id: number
    }
  ]
}
payment_source?: {
  [payment_method_id: number]: {
    number: string
    month: string
    year: string
    verification_value: string
    name: string
  }
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__
```ts
// Logged in user
const response = await client.cart.orderUpdate({ bearerToken }, { order: {...} })

// or guest user
const response = await client.cart.orderUpdate({ orderToken }, { order: {...} })
```

### `orderNext`

Goes to the next Checkout step.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.orderNext({ bearerToken })

// or guest user
const response = await client.cart.orderNext({ orderToken })
```

### `advance`

> Advances Checkout to the furthest Checkout step validation allows, until the Complete step.

Reviewer: ```suggestion
Advances Checkout to the furthest Checkout step allowed by validation, until the Complete step.
```

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.advance({ bearerToken })

// or guest user
const response = await client.cart.advance({ orderToken })
```

### `complete`
Completes the Checkout.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.complete({ bearerToken })

// or guest user
const response = await client.cart.complete({ orderToken })
```

### `addStoreCredits`
Adds Store Credit payments if a user has any.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Parameters schema:__
```ts
{
  amount: number
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.addStoreCredits({ bearerToken }, { amount: 100 })

// or guest user
const response = await client.cart.addStoreCredits({ orderToken }, { amount: 100 })
```

### `removeStoreCredits`

> Remove Store Credit payments if any applied.

Reviewer: ```suggestion
Removes Store Credit payments if any applied.
```

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.removeStoreCredits({ bearerToken })

// or guest user
const response = await client.cart.removeStoreCredits({ orderToken })
```

### `paymentMethods`
Returns a list of available Payment Methods.

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.cart.paymentMethods({ bearerToken })

// or guest user
const response = await client.cart.paymentMethods({ orderToken })
```

### `shippingMethods`

> Returns a list of available Shipping Rates for Checkout. Shipping Rates are grouped against Shipments. Each checkout cna have multiple Shipments eg. some products are available in stock and will be send out instantly and some needs to be backordered.

Reviewer: ```suggestion
Returns a list of available Shipping Rates for Checkout. Shipping Rates are grouped against Shipments. Each checkout can have multiple Shipments e.g. some products are available in stock and will be send out instantly and some need to be backordered.
```

__Required token:__ [Bearer token](#bearer-token) or [Order token](#order-token)

__Optional parameters schema:__
```ts
{
  params?: {
    include?: string
  }
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
// Logged in user
const response = await client.checkout.shippingMethods({ bearerToken }, {
  include: 'shipping_rates'
})

// or guest user
const response = await client.checkout.shippingMethods({ orderToken }, {
  include: 'shipping_rates'
})
```

## [Products](https://guides.spreecommerce.org/api/v2/storefront/#tag/Products)
Returns a list of Products.

### `list`

__Optional parameters schema:__
```ts
{
  include?: string
  fields?: {
    [key: string]: string
  }
  filter?: {
    [key: string]: number
  }
  sort?: string
  page?: number
  per_page?: number
}
```
__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
const response = await client.products.list()
```

### `show`

__Optional parameters schema:__
```ts
{
  id: string
  params?: {
    include: string
    fields: {
      [key: string]: string
    }
  }
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
const response = = await client.products.show('123')
```

## [Taxons](https://guides.spreecommerce.org/api/v2/storefront/#tag/Taxons)

### `list`
Returns a list of Taxons.

__Optional parameters schema:__

```ts
{
  include?: string
  fields?: {
    [key: string]: string
  }
  filter?: {
    [key: string]: number
  }
  page?: number
  per_page?: number
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
const response = await client.taxons.list()
```

### `show`
Returns a single Taxon.

__Optional parameters schema:__

```ts
{
  id: string
  params?: {
    include: string
    fields: {
      [key: string]: string
    }
  }
}
```

__Success response schema:__ [Success schema](#success-schema)

> __Failed response schema:__ [Error schema](#error-schema)

Reviewer: ```suggestion
__Failure response schema:__ [Error schema](#error-schema)
```

__Example:__

```ts
const products = await client.taxons.show('1')
```

## Checkout Flow

### One step

```ts

> await client.cart.create()

Reviewer: ```
const cartCreateResponse = await client.cart.create()

const orderToken = cartCreateResponse.success().data.attributes. token
await client.cart.addItem({ orderToken }, { variant_id: '1' })
  
const shipping = (await client.checkout.shippingMethods({ orderToken })).success()
const payment = (await client.checkout.paymentMethods({ orderToken })).success()

// pick a shipping and payment method

await client.checkout.orderUpdate({ orderToken }, { order: {...} })
await client.checkout.complete({ orderToken })
```

await client.cart.addItem({ orderToken }, { variant_id: '1' })

const shipping = await client.checkout.shippingMethods({ orderToken })

const payment = await client.checkout.paymentMethods({ orderToken })

let order = await client.checkout.orderUpdate({ orderToken }, { order: {...} })

order = await client.checkout.complete({ orderToken })
```

### Three steps

> ```ts

Reviewer: Suggest updating the same way as "One step" above.
